packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
     "MQ Dodongo Cavern Pot Green Room 2": "true"
     "MQ Dodongo Cavern Pot Green Room 3": "true"
     "MQ Dodongo Cavern Pot Green Room 4": "true"
-    "MQ Dodongo Cavern Grass Green Corridor Side Room 1": "(can_use_bow || has_bombflowers || can_use_din) && can_cut_grass"
-    "MQ Dodongo Cavern Grass Green Corridor Side Room 2": "(can_use_bow || has_bombflowers || can_use_din) && can_cut_grass"
+"Dodongo Cavern MQ Poe Room Side Room":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Poe Room": "soul_keese && soul_deku_scrub && can_damage"
+  locations:
+    "MQ Dodongo Cavern GS Poe Room Side": "gs && can_collect_distance"
+    "MQ Dodongo Cavern Grass Green Corridor Side Room 1": "can_cut_grass"
+    "MQ Dodongo Cavern Grass Green Corridor Side Room 2": "can_cut_grass"
 "Dodongo Cavern Bomb Bag Ledge":
   dungeon: DC
   exits:
     "Dodongo Cavern Poe Room": "true"
+    "Dodongo Cavern Main Room Lower": "true"
   locations:
     "MQ Dodongo Cavern Bomb Bag Chest": "true"
-"Dodongo Cavern Skull":
+"Dodongo Cavern Main Room Inside Skull":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Main Room Lower": "event(MQ_DC_OPEN_SKULL)"
+    "Dodongo Cavern Pre-Boss Lobby": "true"
+"Dodongo Cavern Pre-Boss Lobby":
   dungeon: DC
   exits:
-    "Dodongo Cavern Main": "true"
-    "Dodongo Cavern Boss": "true"
+    "Dodongo Cavern Main Room Inside Skull": "true"
+    "Dodongo Cavern Boss": "event(MQ_DC_BOSS_SWITCH)"
+    "Dodongo Cavern Boss Loop Start": "true"
+    "Dodongo Cavern Boss Loop Ending Ledge": "is_adult || climb_anywhere || hookshot_anywhere"
   locations:
-    "MQ Dodongo Cavern Chest Under Grave": "true"
-    "MQ Dodongo Cavern GS Near Boss": "gs"
     "MQ Dodongo Cavern Pot Before Boss 1": "true"
     "MQ Dodongo Cavern Pot Before Boss 2": "true"
+"Dodongo Cavern Boss Loop Start":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Pre-Boss Lobby": "true"
+    "Dodongo Cavern Boss Loop After Fire Wall": "is_adult || climb_anywhere || hookshot_anywhere || event(MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_BEFORE_FIRE)"
+  events:
+    MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_BEFORE_FIRE: "has_bombs || has_bombchu"
+  locations:
     "MQ Dodongo Cavern Pot Before Boss Loop 1": "true"
     "MQ Dodongo Cavern Pot Before Boss Loop 2": "true"
-    "MQ Dodongo Cavern Pot Before Boss Loop 3": "true"
-    "MQ Dodongo Cavern Pot Before Boss Loop 4": "true"
+"Dodongo Cavern Boss Loop After Fire Wall":
+  dungeon: DC
+  exits:
+    "Dodongo Cavern Boss Loop Side Room": "true"
+    "Dodongo Cavern Boss Loop Start": "event(MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_AFTER_FIRE) || climb_anywhere || hookshot_anywhere"
+    "Dodongo Cavern Boss Loop After Armos Wall": "climb_anywhere || hookshot_anywhere || has_fire || has_explosives"
+  events:
+    MQ_DC_BOSS_LOOP_FIRE_WALL_CRYSTAL_AFTER_FIRE: "has_weapon || has_bombs || can_use_mask_blast || has_bombchu || can_use_bow || can_use_slingshot || can_hookshot || can_use_sticks || can_hammer || can_boomerang" #Many ways to hit crystal switch
```

Comment:
Seems like this could really use that crystal switch macro

Reply:
I agree. xD